azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -692,13 +693,12 @@ private Status runJob() {
     try {
       job.run();
     } catch (Throwable e) {
-
       if (props.getBoolean("job.succeed.on.failure", false)) {
         finalStatus = changeStatus(Status.FAILED_SUCCEEDED);
         logError("Job run failed, but will treat it like success.");
         logError(e.getMessage() + " cause: " + e.getCause(), e);
       } else {
-        finalStatus = changeStatus(Status.FAILED);
+        finalStatus = (e instanceof OutOfMemoryException) ? changeToFailureStatusWithCause(ExecutableNode.FailureCause.OOM) : changeStatus(Status.FAILED);
```

Comment:
Would it be simpler to detect low memory situation in azkaban.execapp.JobRunner#run and retry after sleeping for some time?